servers/physics_3d/space_3d_sw.cpp

```diff
@@ -332,7 +332,7 @@ bool PhysicsDirectSpaceState3DSW::cast_motion(const RID &p_shape, const Transfor
 			best_first = false;
 			if (col_obj->get_type() == CollisionObject3DSW::TYPE_BODY) {
 				const Body3DSW *body = static_cast<const Body3DSW *>(col_obj);
-				r_info->linear_velocity = body->get_linear_velocity() + (body->get_angular_velocity()).cross(body->get_transform().origin - closest_B);
+				r_info->linear_velocity = body->get_linear_velocity() + (body->get_angular_velocity()).cross(closest_B - body->get_transform().origin);
```

Comment:
In order to also take the center of mass into account, the correct calculation should be actually:
```
Vector3 rel_vec = closest_B - body->get_transform().origin - body->get_center_of_mass();
r_info->linear_velocity = body->get_linear_velocity() + body->get_angular_velocity().cross(rel_vec);
```
Same for the other cases.